antenna/ext/gcs/crashstorage.py

```diff
@@ -0,0 +1,180 @@
+    https://googleapis.dev/python/google-api-core/latest/auth.html:
+
+    - If you're running in a Google Virtual Machine Environment (Compute Engine, App
+      Engine, Cloud Run, Cloud Functions), authentication should "just work".
+    - If you're developing locally, the easiest way to authenticate is using the `Google
+      Cloud SDK <http://cloud.google.com/sdk>`_::
+
+        $ gcloud auth application-default login
+
+    - If you're running your application elsewhere, you should download a `service account
+      <https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating>`_
+      JSON keyfile and point to it using an environment variable::
+
+        $ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"
+
+
+    **Local emulator**
+
+    If you set the environment variable ``STORAGE_EMULATOR_HOST=http://host:port``,
+    then this will connect to a local GCS emulator.
+
+
+    """
+
+    class Config:
+        bucket_name = Option(
+            doc=(
+                "Google Cloud Storage bucket to save to. Note that the bucket must "
+                "already have been created."
+            ),
+        )
+
+    def __init__(self, config):
+        self.config = config.with_options(self)
+        self.bucket = self.config("bucket_name")
+
+        if emulator := os.environ.get("STORAGE_EMULATOR_HOST"):
+            logger.debug(
+                "STORAGE_EMULATOR_HOST detected, connecting to emulator: %s",
+                emulator,
+            )
+            self.client = storage.Client(
+                credentials=AnonymousCredentials(),
+                project="test",
+            )
+        else:
+            self.client = storage.Client()
+
+        register_for_verification(self.verify_write_to_bucket)
+
+    def _save_file(self, path, data):
+        """Save a single file to GCS.
+
+        :arg str path: the path to save to
+        :arg bytes data: the data to save
+
+        """
+        bucket = self.client.get_bucket(self.bucket)
+        blob = bucket.blob(path)
+        blob.upload_from_string(data)
+
+    def verify_write_to_bucket(self):
+        """Verify GCS bucket exists and can be written to."""
+        self._save_file(generate_test_filepath(), b"test")
+
+    def check_health(self, state):
+        """Check GCS connection health."""
+        try:
+            # get the bucket to verify GCS is up and we can connect to it.
+            self.client.get_bucket(self.bucket)
+        except Exception as exc:
+            state.add_error("GcsCrashStorage", repr(exc))
+
+    def _get_raw_crash_path(self, crash_id):
+        return "v1/raw_crash/{date}/{crash_id}".format(
+            date=get_date_from_crash_id(crash_id),
+            crash_id=crash_id,
+        )
+
+    def _get_dump_names_path(self, crash_id):
+        return f"v1/dump_names/{crash_id}"
+
+    def _get_dump_name_path(self, crash_id, dump_name):
+        if dump_name in (None, ""):
+            dump_name = "dump"
```

Comment:
This is probably not what we want. Previously `None`, `""`, and `upload_file_minidump` would all get saved as `"dump"`--so they're all treated as the same thing.

With this change, we create a new scenario where we have `"upload_file_minidump"` but also have `"dump"`. So previously if the multipart/form-data item didn't have a name, we assumed it was `"upload_file_minidump"` and treated it like that. Now if it doesn't have a name, it gets treated like `"dump"` which isn't a thing.

We should change this to treat them all like `"upload_file_minidump"`.

```suggestion
            dump_name = "upload_file_minidump"
```